playbooks/ai-on-openlab/pre.yaml

```diff
@@ -0,0 +1,93 @@
+- name: AI On Openlab Pre Job
+  hosts: all
+  become: yes
+  tasks:
+    - name: Set Default Hardware type Parameters
+      set_fact:
+        user_file: "user_data"
+      when: not is_gpu_enabled
+
+    - name: Split Hardware type Parameters for GPU
+      set_fact:
+        user_file: "user_data_gpu"
+      when: is_gpu_enabled
+
+    - name: Shape login Demo User
+      shell:
+        cmd: |
+          set -x
+          set -e
+
+          useradd demo -p 'demo'
+          echo 'demo:demo' | chpasswd
+          usermod -s /bin/bash demo
+          mkdir /home/demo
+          chown demo:demo /home/demo
+          sed -i 's/PasswordAuthentication no/PasswordAuthentication yes/' /etc/ssh/sshd_config
+          service sshd restart
+        executable: /bin/bash
+
+    - name: Upload Environment Info to Web
+      shell:
+        cmd: |
+          apt install -y python-pip
+          pip install pygithub
+
+          source_file=`ls "{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/" | grep moo-job`
+          job_name=$(echo $source_file | cut -d . -f1)
+          job_name=${job_name/_sleep/}
+          if {{ env_keeping | lower }};then
+            wget https://raw.githubusercontent.com/moo-ai/moo-ai.github.io/master/env_info/task_list_sleep.json -O env.json
+          else
+            wget https://raw.githubusercontent.com/moo-ai/moo-ai.github.io/master/env_info/task_list.json -O env.json
+          fi
+          cat > 'json_output.py' <<EOF
+          import json
+          exist_data = None
+          with open('env.json') as json_file:
+              exist_data = json.load(json_file)
+
+          ip_addr = "{{ nodepool.public_ipv4 }}"
+
+          exist_data["$job_name"] = {
+              "patch_set": "{{ zuul.patchset }}",
+          }
+          if '{{ env_keeping | lower }}' == 'true':
```

Comment:
is it an string variable? not boolean?

Reply:
I'll move python script to moo repo. No need for these code then.